tests/e2e/test_create_view_edit_grant.py

```diff
@@ -0,0 +1,62 @@
+import uuid
+
+from playwright.sync_api import Page, expect
+
+from tests.e2e.pages import AllGrantsPage, ChangeGrantNamePage, GrantDashboardPage, GrantSettingsPage, NewGrantPage
+
+
+def test_create_view_edit_grant_success(page: Page, domain: str):
+    all_grants_page = AllGrantsPage(page, domain)
+    all_grants_page.navigate()
+    expect(all_grants_page.title).to_be_visible()
+
+    # Set up new grant
+    new_grant_page: NewGrantPage = all_grants_page.click_set_up_new_grant()
+    expect(new_grant_page.title).to_be_visible()
+
+    new_grant_name = f"E2E {uuid.uuid4()}"
+    new_grant_page.complete_grant_name(new_grant_name)
+    all_grants_page: AllGrantsPage = new_grant_page.submit_new_grant_form()
```

Comment:
```suggestion
    all_grants_page = new_grant_page.submit_new_grant_form()
```

I think these manual type hints shouldn't be needed, because the methods set their return types